modules/puppetca/puppetca_main.rb

```diff
@@ -47,11 +46,12 @@ def disable certname
     def autosign certname
       FileUtils.touch(autosign_file) unless File.exist?(autosign_file)
 
-      autosign = File.open(autosign_file, File::RDWR)
-      # Check that we don't have that host already
-      found = autosign.readlines.find { |line| line.chomp == certname }
-      autosign.puts certname unless found
-      autosign.close
+      open(autosign_file, File::RDWR) do |autosign|
+        # Check that we don't have that host already
+        found = false
+        autosign.each_line { |line| found = true if line.chomp == certname }
```

Comment:
That would be https://github.com/ekohl/smart-proxy/commit/af0a3a4fbbc81f438726406dfedb3cacbf5abce5. I also took the liberty to add a testcase for it.